### DISCUSSION TITLE
Calculating pos_ydir from matsimnibs

### DISCUSSION THREAD

**Discussion by wys5188** [simnibs/simnibs#318] [8. Simulation workflows] [answered]
Hello,
We want to use simnibs to estimate the TMS E-field post-hoc using the "TriggerMarkers" file. We are able to import this file into a tms_list using the code below. However, we need to calculate `pos_ydir` from the 4x4 matsimnibs matrix. We know that `centre` is the last column of the matrix, but do not know how to derive `pos_ydir` from the remaining 3x3 matrix. We are using a MagVenture Cool B-70 coil. Our particular pipeline is expecting the `centre` and  `pos_ydir` vectors rather than the 4x4 matrix. Is there a simple way to get that information from the matrix? We've been able to do the reverse (get `matsimnibs` from `centre` and `pos_ydir`) with the simnibs_gui.

```
from simnibs import sim_struct, localite

s = sim_struct.SESSION()

fn = ".../TMSTrigger/TriggerMarkers_Coil0_XXX.xml"

tms_list = localite().read(fn)  # read all TriggerMarkers from file and return as TMSLIST()
s.add_tmslist(tms_list)

print(tms_list.pos[0].didt) 
print(tms_list.pos[0].matsimnibs)  
print(tms_list.pos[0].centre)  
print(tms_list.pos[0].pos_ydir)
```

This is the output:
```
python test_simnibs.py 
42000000.0
[[-6.25310075e-01  5.16387488e-01  5.85024118e-01  3.78978130e+01]
 [-8.36362255e-02  7.01017412e-01 -7.08167718e-01 -3.10586757e+01]
 [-7.75831320e-01 -4.91772811e-01 -3.95179973e-01 -1.03718364e+02]
 [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  1.00000000e+00]]
None
None
```

In this case, we know that `centre` is [37.9, -31.1, -103.7]. But how do we calculate `pos_ydir`?

Thank you,
Wendy

**Comment by TorgeW** [comment, marked as answer]
```
center  = matsimnibs[:3,3]
pos_ydir = center + 20 * matsimnibs[:3, 1]
```

This should transform a matsimnibs to a center and pos_ydir

**Comment by eburkevt** [comment]
Why the factor of 20 in the following?

pos_ydir = center + 20 * matsimnibs[:3, 1]

I'm asking because our lab has a similar need to determine the orientation of our TMS coil relative to the T1w image coordinate system.

**Reply by TorgeW** [reply]
20 is not critical, its just to make sure the pos_ydir is not to close and not to far in relation to a human head. 
The pos_ydir position will then be placed 2 cm away from the center position.

**Reply by eburkevt** [reply]
Okay, great. Thank you for the information.
